hnn_core/opt_toy_example/general.py

```diff
@@ -0,0 +1,365 @@
+"""Parameter optimization."""
+
+# Authors: Carolina Fernandez <cxf418@miami.edu>
+#          Nick Tolley <nicholas_tolley@brown.edu>
+#          Ryan Thorpe <ryan_thorpe@brown.edu>
+#          Mainak Jas <mjas@mgh.harvard.edu>
+
+import numpy as np
+
+from metrics import _rmse_evoked  # change path***
+
+from skopt import gp_minimize
+from scipy.optimize import fmin_cobyla
+
+
+class Optimizer:
+    def __init__(self, net, constraints, set_params, solver, obj_fun,
+                 tstop, scale_factor=1., smooth_window_len=None):
+        if net.external_drives:
+            raise ValueError("The current Network instance has external " +
+                             "drives, provide a Network object with no " +
+                             "drives.")
+        self.net = net
+        self.constraints = constraints
+        self._set_params = set_params
+        # Optimizer method
+        if solver == 'bayesian':
+            self._assemble_constraints = _assemble_constraints_bayesian
+            self._run_opt = _run_opt_bayesian
+        elif solver == 'cobyla':
+            self._assemble_constraints = _assemble_constraints_cobyla
+            self._run_opt = _run_opt_cobyla
+        else:
+            raise ValueError("solver must be 'bayesian' or 'cobyla'")
+        # Response to be optimized
+        if obj_fun == 'evoked':
+            self.obj_fun = _rmse_evoked
+        else:
+            raise ValueError("obj_fun must be 'evoked'")
+        self.scale_factor = scale_factor
+        self.smooth_window_len = smooth_window_len
+        self.tstop = tstop
+        self.net_ = None
+        self.obj = list()
+        self.opt_params = None
+        self.max_iter = 200
+
+    def __repr__(self):
+        class_name = self.__class__.__name__
```

Comment:
can we make it more informative? maybe printing some of the attributes of the optimizer + whether the optimizer has already been fit would be useful information